packages/slate/src/changes/with-schema.js

```diff
@@ -47,7 +47,9 @@ Changes.normalizeNodeByKey = (change, key) => {
   if (!ancestors) return
 
   ancestors.forEach(ancestor => {
-    normalizeNode(change, ancestor, schema)
+    if (change.value.document.getDescendant(ancestor.key)) {
```

Comment:
In this case, the following `normalizeNode` will call `getDescendant(ancestor.key)`.  Because of the memorizing in slate, the overall effect is that this loop is not slowed down by `getDescendant(ancestor.key)`.